numseq/fib.py

```diff
@@ -0,0 +1,16 @@
+def memoize(func):
+    memo = {}
+
+    def inner(x):
+        if x not in memo:
+            memo[x] = func(x)
+        return memo[x]
+    return inner
+
+
+@memoize
+def fib(n):
```

Comment:
Careful here -- you are not handling the case of n <= 0.  This is causing an infinite loop because you are trying to evaluate fib(n) of negative numbers which is undefined.